apps/block_scout_web/lib/block_scout_web/views/generic_pagination_helpers.ex

```diff
@@ -0,0 +1,56 @@
+defmodule BlockScoutWeb.GenericPaginationHelpers do
+  @moduledoc """
+  Helpers for handling pagination
+  """
+
+  def next_page_path(conn, params, paging_options, total_item_count, path_fun) do
+    if paging_options.page_number * paging_options.page_size >= total_item_count do
+      nil
+    else
+      path_fun.(conn, Map.merge(params, %{paging_options | page_number: paging_options.page_number + 1}))
+    end
+  end
+
+  def prev_page_path(conn, params, paging_options, path_fun) do
+    if paging_options.page_number == 1 do
+      nil
+    else
+      path_fun.(conn, Map.merge(params, %{paging_options | page_number: paging_options.page_number - 1}))
+    end
+  end
+
+  def first_page_path(conn, params, paging_options, path_fun) do
+    if paging_options.page_number == 1 do
+      nil
+    else
+      path_fun.(conn, Map.merge(params, %{paging_options | page_number: 1}))
+    end
+  end
+
+  def last_page_path(conn, params, paging_options, total_item_count, path_fun) do
+    last_page_number = ceil(total_item_count / paging_options.page_size)
+
+    if paging_options.page_number == last_page_number do
+      nil
+    else
+      path_fun.(conn, Map.merge(params, %{paging_options | page_number: last_page_number}))
+    end
+  end
+
+  def current_page(paging_options), do: paging_options.page_number
+
+  def sort_path(conn, params, paging_options, field, default_order_dir, path_fun) do
+    path_fun_params =
+      params
+      |> Map.merge(paging_options)
+      |> Map.merge(
```

Comment:
You are creating a map with a single key + value pair within this function call, it seems unusual to call `merge` consecutively. It would be better to move this out and set the values directly.

I'm trying to suggest some changes but I'm finding it hard to understand what is being done here. If `order_field` matches the parameter `field`, and `order_dir` is `desc`, it will invert it? So this isn't actually getting the sort path but toggling the sort order?